src/response.rs

```diff
@@ -250,6 +250,30 @@ impl Response {
     /// This consumes the body. Trying to read more, or use of `response.json()`
     /// will return empty values.
     pub fn text(&mut self) -> ::Result<String> {
+        self.text_with_charset(UTF_8)
+    }
+
+    /// Get the response text given a specific encoding.
+    ///
+    /// This method decodes the response body with BOM sniffing
+    /// and with malformed sequences replaced with the REPLACEMENT CHARACTER.
+    /// Encoding is determinated from the `charset` parameter of `Content-Type` header,
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # extern crate reqwest;
+    /// # fn run() -> Result<(), Box<::std::error::Error>> {
+    /// let content = reqwest::get("http://httpbin.org/range/26")?.text_with_charset(encoding_rs::UTF_8)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// This consumes the body. Trying to read more, or use of `response.json()`
+    /// will return empty values.
+    pub fn text_with_charset(&mut self, enc: &'static Encoding) -> ::Result<String> {
```

Comment:
This exposes the `encoding_rs` dependency publicly in reqwest's API. The only downside to this is that if `encoding_rs` releases a breaking change, reqwest cannot upgrade to it without that *also* being a breaking change.

What if this method instead just accepted a name via string?

Reply:
You are right. I will make that changes real fast. Thanks for the quick response.

Reply:
Done